lib/src/components/NxFilterInput/NxFilterInput.tsx

```diff
@@ -18,18 +18,16 @@ export { Props } from './types';
 
 const NxFilterInput = forwardRef<HTMLDivElement, Props>(
     function NxFilterInput(props, ref) {
-      const classes = classnames('nx-filter-input', props.className),
+      const { className: classNameProp, searchIcon, ...otherProps } = props,
+          className = classnames('nx-filter-input', classNameProp),
 
           // just in case these props get passed in, avoid passing them to NxTextInput as they would cause
           // malfunction
-          otherProps = omit(['validatable', 'validationErrors', 'type'], props),
-          filterIcon = <NxFontAwesomeIcon icon={faFilter} className="nx-icon--filter-icons" />;
+          cleanedProps = omit(['validatable', 'validationErrors', 'type'], otherProps),
+          filterIcon = searchIcon ? faSearch : faFilter,
```

Comment:
> I'm curious: in what case would these props get passed in? Is it in the context where it's wrapped within an NxForm?

Is this referring to the `searchIcon` prop or the props being omitted on the line above that?

Reply:
> > I'm curious: in what case would these props get passed in? Is it in the context where it's wrapped within an NxForm?
> 
> Is this referring to the `searchIcon` prop or the props being omitted on the line above that?

Oh I'm sorry, I meant the props being omitted.

Reply:
> Oh I'm sorry, I meant the props being omitted.

There isn't any specific case where they _would_ be passed in, but it would be possible in a non-typescript project for someone to try to pass them in.  And if they did so I believe that would break this component if not for this check.